### ISSUE TITLE
Errors when executing `pip install comet`

### ISSUE THREAD

**Issue by jmcquaid87** [jdswinbank/Comet#45]
Cannot properly install comet via pip.  I executed the following:

    $ virtualenv comet
    $ . comet/bin/activate
    $ pip install comet

Here is the log file:

[cometlog.txt](https://github.com/jdswinbank/Comet/files/543314/cometlog.txt)

**Comment by jdswinbank** [comment]
Looks like the problem is that you don't have the Python headers installed, so you can't compile extension modules. Comet doesn't need compiling itself, but its dependencies (Twisted, lxml) do.

Best way to solve this depends on your operating system and how you've installed Python. On a Debian/Ubuntu based system—which [I gather](https://en.wikipedia.org/wiki/Raspberry_Pi#Operating_systems) is the default on a Raspberry Pi—you probably want to run "apt-get install python-dev" or similar.

Does that help?

**Comment by jmcquaid87** [comment]
I ran `apt-get install python-dev` and it ran successfully.  I then ran `pip install comet`, and there were again errors, but they look different than the ones from before.  Here is the log file:
[cometlog.txt](https://github.com/jdswinbank/Comet/files/543359/cometlog.txt)

**Comment by jdswinbank** [comment]
I saw a couple of different logs fly by in the above comment: I guess you edited it to replace one with the other? I saw a couple of different issues, though:

- You'll need to ``apt-get install libxml2-dev`` (or similar, depending on your operating system), in order to be able to compile lxml. You'll probably also need ``libxslt-dev``.
- It looks like you're using Python 3 (based on ``lxml>=2.3 in /usr/lib/python3/dist-packages``). The pip-installable version of Comet doesn't currently support Python 3; only the bleeding-age master branch does. If you need to install a version that works with Python 3, see https://github.com/jdswinbank/Comet/issues/44#issuecomment-253626173 for details.

**Comment by jmcquaid87** [comment]
I made sure to run `sudo apt-get install libxml2-dev` as well as `libxslt-dev`.  This time, I ran `pip2.7 install comet`, because when I run `python --version` it shows i have 2.7.9, so I wanted to make sure pip installs to Python 2.7.  I attached my most recent log; it doesn't have an error but it now just hangs at a certain point during the install.
[cometlog.txt](https://github.com/jdswinbank/Comet/files/544209/cometlog.txt)
